### DISCUSSION TITLE
RAM loading on SDRAM

### DISCUSSION THREAD

**Discussion by Sathishkumar-rk** [zephyrproject-rtos/zephyr#48242] [Q&A]
Hi @DerekSnell 

I want to use the SDRAM for zephyr application running, below is detailed requirement for my project using mimxrt1160_evk_cm7.

MCUBoot will copy the latest image from primary or secondary slot based on image version using RAM Loading feature and then I want run the application from RAM location.

I was able to copy the content in SDRAM location (0x80020000) but while execution it still running from flash (expected behavior is running from SDRAM location).

Image details on slot-0 & slot-1:
<img width="356" alt="Screenshot 2022-07-19 170435" src="https://user-images.githubusercontent.com/105291276/180719434-26c510e1-0a72-4308-ba03-7587b00ca6cb.png">

Whenever I run the program it always run from slot-0 program, Is that I'm missing some setting in zephyr application or mcuboot?
MCUBoot configuration:
![image](https://user-images.githubusercontent.com/105291276/180720538-890c4ed3-cb0d-4929-9a33-eb323dd3384a.png)

**Comment by DerekSnell** [comment]
Hi @Sathishkumar-rk ,

If you want your Zephyr app to execute from somewhere other than flash, you need to configure the app to execute from that other memory.  I suspect you built with the default CONFIG_CODE_FLEXSPI=y, which builds for flash.  The app may be copied from flash to SDRAM, but when the app executes a branch instruction, the address will still be in flash, and it will jump back to flash and execute from there.

You can build the code for SDRAM using CONFIG_CODE_SEMC=y.  However, when I test with this, I see when z_bss_zero() is called in the startup code, it trashes the code memory in SDRAM.  So it seems the linker settings for this RT1160 are not configured for CONFIG_CODE_SEMC=y, and allow the BSS section to overlap code in SDRAM.

You can create an issue to report this, and it will be assigned to NXP.  You can also link back to this discussion to detail the problem.

Thanks

**Reply by DerekSnell** [reply]
linking to the new issue https://github.com/zephyrproject-rtos/zephyr/issues/50652
